nixos/modules/services/mail/postal.nix

```diff
@@ -0,0 +1,530 @@
+            };
+
+            default_health_server_bind_address = mkOption {
+              type = types.str;
+              default = "127.0.0.1";
+              description = "The default bind address for the worker health server to listen on";
+            };
+          };
+
+          main_db = {
+            host = mkOption {
+              type = types.str;
+              default = "localhost";
+              description = "Hostname for the main MariaDB server";
+            };
+
+            port = mkOption {
+              type = types.port;
+              default = 3306;
+              description = "The MariaDB port to connect to";
+            };
+
+            username = mkOption {
+              type = types.str;
+              default = cfg.user;
+              defaultText = literalExpression "config.${opt.user}";
+              description = "The MariaDB username";
+            };
+
+            database = mkOption {
+              type = types.str;
+              default = "postal";
+              description = "The MariaDB database name";
+            };
+
+            encoding = mkOption {
+              type = types.str;
+              default = "utf8mb4";
+              description = "The encoding to use when connecting to the MariaDB database";
+            };
+          };
+
+          message_db = {
+            host = mkOption {
+              type = types.str;
+              default = "localhost";
+              description = "Hostname for the MariaDB server which stores the mail server databases";
+            };
+
+            port = mkOption {
+              type = types.port;
+              default = 3306;
+              description = "The MariaDB port to connect to";
+            };
+
+            username = mkOption {
+              type = types.str;
+              default = cfg.user;
+              defaultText = literalExpression "config.${opt.user}";
+              description = "The MariaDB username";
+            };
+
+            encoding = mkOption {
+              type = types.str;
+              default = "utf8mb4";
+              description = "The encoding to use when connecting to the MariaDB database";
+            };
+
+            database_name_prefix = mkOption {
+              type = types.str;
+              default = "postal";
+              description = "The MariaDB prefix to add to database names";
+            };
+          };
+
+          smtp_server = {
+            default_port = mkOption {
+              type = types.port;
+              default = if cfg.settings.smtp_server.tls_enabled then 587 else 25;
+              defaultText = literalExpression "if config.services.postal.settings.smtp_server.tls_enabled then 587 else 25";
+              description = "The default port the SMTP server should listen on unless overriden by the PORT environment variable";
+            };
+
+            default_bind_address = mkOption {
+              type = types.str;
+              default = "::";
```

Comment:
I'm not too happy with the mixture of v4 only binds everywhere except here, can you look into binding to v6 and v4 for binds or use unix sockets where possible?